extensions/3rdparty/LyricWiki/LyricFind/LyricFindHooks.class.php

```diff
@@ -111,4 +114,29 @@ static public function onParserBeforeStrip(Parser $parser, &$text, &$strip_state
 		return true;
 	}
 
+	public static function isWebCrawler( $userAgent ) {
+		// A list of some common words used only for bots and crawlers.
+		$crawlers = [
+			'bot',
+			'slurp',
+			'crawler',
+			'spider',
+			'curl',
+			'facebook',
+			'fetch',
+			'ruby',
+			'python',
+			'perl/lyrics::fetcher::lyricwiki',
+			'mozilla/5.0 (windows; u; windows nt 5.1; en-us; rv:1.8.1.6) gecko/20070725 firefox/2.0.0.6',
+			'mozilla/5.0 (x11; linux x86_64) applewebkit/535.19 (khtml, like gecko) chrome/18.0.1025.142 safari/535.19'
```

Comment:
Who will monitor and maintain this blacklist? 😕

Reply:
I agree, however before we do any change to this logic I'd like to know what are the reasons of asking LyricFind if we can create page with given title. Do you know somebody who could remember this? Or maybe some articles in Confluence? Maybe we could get rid of this call to api from `isPageBlockedViaApi()`.

Reply:
`isPageBlockedViaApi` checks for a 206 response code.  According to [the docs](http://www.lyricfind.com/documentation/?key=eyJkb2N1bWVudCI6IndlYl9zZXJ2aWNlIiwib3B0aW9ucyI6eyJwcmVtaXVtIjoiZmFsc2UiLCJvbGRrZXl0eXBlcyI6ImZhbHNlIn19#the-lyric-display-api) that means:

> Lyric is not available because of a publisher specific block. No data is returned. 

If a user attempts to create a new lyric page, and the API returns this code, we have no license to host those lyrics on our site, and thus there's a potential for copyright infringement.

There's still an open question as to whether or not we can perform this check without it being tracked as a "view" which we must pay for.